contracts/message/messagebus/MessageBusReceiver.sol

```diff
@@ -412,20 +440,34 @@ contract MessageBusReceiver is Ownable {
             );
     }
 
-    function executeMessage(MsgDataTypes.RouteInfo calldata _route, bytes calldata _message)
+    function executeMessage(MsgDataTypes.Route memory _route, bytes calldata _message)
         private
         returns (IMessageReceiverApp.ExecutionStatus)
     {
         uint256 gasLeftBeforeExecution = gasleft();
-        (bool ok, bytes memory res) = address(_route.receiver).call{value: msg.value}(
-            abi.encodeWithSelector(
-                IMessageReceiverApp.executeMessage.selector,
-                _route.sender,
-                _route.srcChainId,
-                _message,
-                msg.sender
-            )
-        );
+        bool ok;
+        bytes memory res;
+        if (_route.senderBytes.length == 0) {
+            (ok, res) = address(_route.receiver).call{value: msg.value}(
+                abi.encodeWithSelector(
+                    bytes4(keccak256(bytes("executeMessage(address,uint64,bytes,address)"))),
+                    _route.sender,
+                    _route.srcChainId,
+                    _message,
+                    msg.sender
+                )
+            );
+        } else {
+            (ok, res) = address(_route.receiver).call{value: msg.value}(
+                abi.encodeWithSelector(
+                    bytes4(keccak256(bytes("executeMessage(bytes,uint64,bytes,address)"))),
+                    _route.sender,
```

Comment:
should use '_route.senderBytes' instead of '_route.sender'?

Reply:
fixed